gcc/brig/brigfrontend/brig-to-generic.cc

```diff
@@ -893,3 +958,22 @@ dump_function (FILE *dump_file, brig_function *f)
       fprintf (dump_file, "\n");
     }
 }
+
+/* Records use of the BRIG_REG as a TYPE in the current function.  */
+
+void
+brig_to_generic::add_reg_used_as_type (const BrigOperandRegister &brig_reg,
+				       tree type)
+{
+  gcc_assert (m_cf);
+  reg_use_info &info
+    = m_fn_regs_use_index[m_cf->m_name][gccbrig_hsa_reg_hash (brig_reg)];
+
+  if (info.m_type_refs_lookup.count (type))
```

Comment:
Is the whole construct needed here? Cannot you rely on the default construction of elements in the map and just omit the if and increment unconditionally?

Reply:
No, the map hold index to m_type_ref vector. If the map does not have the value for the key there is no corresponding entry in the vector either. The map was earlier std::map<tree,std::pair<tree, size_t> > instead of two separate containers, but this made 'register' variable type selection undeterministic.